lib/chef/chef_fs/file_system/repository/nodes_dir.rb

```diff
@@ -30,6 +31,24 @@ class NodesDir < Repository::Directory
           def make_child_entry(child_name)
             Node.new(child_name, self)
           end
+
+          def create_child(child_name, file_contents = nil)
+            child = super
+            File.chmod(0600, child.file_path)
```

Comment:
I elected to skip an `else` since `File.chmod` is a no-op on Windows. I'm fine either way though if we want to put in in an else.